pkg/proxy/ipvs/proxier.go

```diff
@@ -1672,14 +1672,17 @@ func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, curre
 	for cs := range currentServices {
 		svc := currentServices[cs]
 		if _, ok := activeServices[cs]; !ok {
-			// This service was not processed in the latest sync loop so before deleting it,
-			okayToDelete := true
 			rsList, _ := proxier.ipvs.GetRealServers(svc)
 
 			// If we still have real servers graceful termination is not done
 			if len(rsList) > 0 {
-				okayToDelete = false
+				continue
+			}
+
+			if proxier.isIPInExcludeCIDRs(svc.Address) {
```

Comment:
Could we move it at the top of the for loop (and perform this check before doing anything else, to make it clear we completely ignore these CIDRs?)

Reply:
Yeah, of course. It makes sense